src/viking.js

```diff
@@ -1,11 +1,125 @@
 // Soldier
-class Soldier {}
+class Soldier {
+    constructor(health, strength) {
+        this.health = health,
+            this.strength = strength
+    }
+
+    attack() {
+        return this.strength
+
+    }
+    receiveDamage(damage) {
+        this.health -= damage
+
+    }
+}
 
 // Viking
-class Viking {}
+class Viking extends Soldier {
+    constructor(name, health, strength) {
+        super(health, strength)
+        this.name = name
+    }
+
+    receiveDamage(damage) {
+        this.health -= damage
+        if (this.health > 0) {
+            return `${this.name} has received ${damage} points of damage`
+        } else {
+            return `${this.name} has died in act of combat`
+
+        }
+    }
+
+    battleCry() {
+        return `Odin Owns You All!`
+    }
+
+}
 
 // Saxon
-class Saxon {}
+class Saxon extends Soldier {
+
+    receiveDamage(damage) {
+        this.health -= damage
+
+        if (this.health > 0) {
+            return `A Saxon has received ${damage} points of damage`
+        } else {
+            return `A Saxon has died in combat`
+
+        }
+
+    }
+}
 
 // War
-class War {}
+class War {
+    constructor() {
+        this.vikingArmy = [],
+            this.saxonArmy = []
+    }
+
+    addViking(Viking) {
+        this.vikingArmy.push(Viking)
+    }
+
+    addSaxon(Saxon) {
+        this.saxonArmy.push(Saxon)
+    }
+
+    vikingAttack() {
+        let saxon = randomIndex(this.saxonArmy)
+        let viking = randomIndex(this.vikingArmy)
+
+        saxon = saxon.receiveDamage(viking.strength)
+
+        this.saxonArmy = cleanDeads(this.saxonArmy)
+
+        return saxon
+
+    }
+
+    saxonAttack() {
+        let saxon = randomIndex(this.saxonArmy)
+        let viking = randomIndex(this.vikingArmy)
+
+        viking = viking.receiveDamage(saxon.strength)
+
+        this.vikingArmy = cleanDeads(this.vikingArmy)
+
+        return viking
+    }
+
+    showStatus() {
+        if (!this.saxonArmy.length) {
+            return "Vikings have won the war of the century!"
+        }
+        if (!this.vikingArmy.length) {
+            return 'Saxons have fought for their lives and survived another day...'
+        }
+        if (this.vikingArmy.length > 0 && this.saxonArmy.length > 0) {
+            return 'Vikings and Saxons are still in the thick of battle.'
+
+        }
+    }
+}
+
+function randomIndex(array) {
```

Comment:
As discussed above, the naming could be altered. For example "getRandomWarrior" could do. Wouldn't it be better to make that function a method of the War class?